src/subtitles-octopus.js

```diff
@@ -108,6 +136,23 @@ var SubtitlesOctopus = function (options) {
         self.ctx = self.canvas.getContext('2d');
         self.bufferCanvas = document.createElement('canvas');
         self.bufferCanvasCtx = self.bufferCanvas.getContext('2d');
+
+        // test for alpha bug, where e.g. WebKit can render a transparent pixel
+        // (with alpha == 0) as non-black which then leads to visual artifacts
+        self.bufferCanvas.width = 1;
+        self.bufferCanvas.height = 1;
+        var testBuf = new Uint8ClampedArray([0, 255, 0, 0]);
+        var testImage = new ImageData(testBuf, 1, 1);
+        self.bufferCanvasCtx.clearRect(0, 0, 1, 1);
+        self.ctx.clearRect(0, 0, 1, 1);
+        var prePut = self.ctx.getImageData(0, 0, 1, 1).data;
+        self.bufferCanvasCtx.putImageData(testImage, 0, 0);
+        self.ctx.drawImage(self.bufferCanvas, 0, 0);
+        var postPut = self.ctx.getImageData(0, 0, 1, 1).data;
+        self.hasAlphaBug = prePut[1] != postPut[1];
```

Comment:
Duplicate? Also again, I'd extract it.

Reply:
Not a duplicate, I rather try to avoid a potential possibility of some code accessing `self.hasAlphaBug` before checking, so I initialize it to `false` in the constructor and compute its real value when a canvas is made